### PR TITLE
create store v2 file interface

### PR DESCRIPTION
**Pull request by walldiss**
This is the foundation of new ~era~ sampling protocol and storage.

### CODE REVIEW

File: share/store/file.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+type File interface {
+	io.Closer
+	Size() int
+	ShareWithProof(ctx context.Context, axisType rsmt2d.Axis, axisIdx, shrIdx int) (share.Share, nmt.Proof, error)
```
Comment (Wondertan) [inline share/store/file.go:16]: Suggestion:
Let's just call it `Share`. The WithProof part would be needed only if there is a version to get a share without a proof.

File: share/store/file.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	Axis(ctx context.Context, axisType rsmt2d.Axis, axisIdx int) ([]share.Share, error)
```
Comment (Wondertan) [inline share/store/file.go:17]: Do we need it? The sampling protocol does not.

File: share/store/file.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	"context"
+	"io"
+
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+type File interface {
```
Comment (Wondertan) [inline share/store/file.go:13]: Please add comments for the File. In general, treat everything we do in the protocol_v2 branch as we do it in main

File: share/store/file.go (hunk 2 above)
Comment (walldiss) [reply share/store/file.go]: Removed for now. Could be added later if needed

**Pull request merged**